cli/reader/reader.go

```diff
@@ -0,0 +1,120 @@
+package reader
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"sync"
+)
+
+// ContentReader defines an interface for reading content from various sources.
+type ContentReader interface {
+	Read() (string, error)
+}
+
+// FileContentReader reads content from a specified file path.
+type FileContentReader struct {
+	FilePath string
+}
+
+// Read reads the content from the file specified in FileContentReader.
+// It reads the entire file content into memory, which is suitable for smaller files.
+func (f FileContentReader) Read() (string, error) {
+	content, err := os.ReadFile(f.FilePath)
+	if err != nil {
+		return "", fmt.Errorf("error reading file '%s': %w", f.FilePath, err)
+	}
+	return string(content), nil
+}
+
+// StdinContentReader reads content from the standard input (stdin).
+type StdinContentReader struct{}
+
+// Read reads the content from stdin.
+// It reads all the data from stdin until EOF, which is useful for piping input.
+func (s StdinContentReader) Read() (string, error) {
+	input, err := io.ReadAll(os.Stdin)
+	if err != nil {
+		return "", fmt.Errorf("error reading from stdin: %w", err)
+	}
+	return string(input), nil
+}
+
+// ReadContentConcurrently reads content from multiple readers concurrently and returns the results.
+// This function utilizes goroutines to perform concurrent reads, improving performance for multiple files.
+func ReadContentConcurrently(readers []ContentReader) ([]string, error) {

```

Comment:
Maybe a variadic here


```suggestion
func ReadContentConcurrently(readers ...ContentReader) ([]string, error) {
```

This way calling the function would be more readable

Reply:
But because of getReaders, it may be pointless

Reply:
> Maybe a variadic here
> 
> This way calling the function would be more readable

I'll work on some refactoring next, which tackles the #30 issue, when I open a PR I may include your suggestion then. Thanks, as always.

Reply:
See PR #37